src/sonic_ax_impl/mibs/ietf/rfc1213.py

```diff
@@ -667,3 +668,34 @@ class InterfacesMIB(metaclass=MIBMeta, prefix='.1.3.6.1.2.1.2'):
     # FIXME Placeholder
     ifSpecific = \
         SubtreeMIBEntry('2.1.22', if_updater, ValueType.OBJECT_IDENTIFIER, lambda sub_id: ObjectIdentifier.null_oid())
+
+class sysNameUpdater(MIBUpdater):
+    def __init__(self):
+        super().__init__()
+        self.db_conn = mibs.init_db()
+        self.hostname = socket.gethostname()
+
+    def reinit_data(self):
+        self.db_conn.connect(self.db_conn.CONFIG_DB)
+        device_metadata = self.db_conn.get_all(self.db_conn.CONFIG_DB, "DEVICE_METADATA|localhost")
```

Comment:
>get_all [](start = 39, length = 7)

Did you test the empty table on DUT? I think device_metadata will be an empty dict instead of None.
Suggest check `device_metadata` instead of `device_metadata is not None` below.

Reply:
Alternatively, you can use `get` function to get `hostname` directly.

---
In reply to: [548318135](https://github.com/Azure/sonic-snmpagent/pull/185#discussion_r548318135) [](ancestors = 548318135)

Reply:
If DEVICE_METADATA table is not present, then get_all returns None. 
Updated code as per suggestion and tested on sonic-vs image.

Reply:
I am okay with the code. However I am confused by your comment "If DEVICE_METADATA table is not present, then get_all returns None".
Could you double check?

---
In reply to: [548356652](https://github.com/Azure/sonic-snmpagent/pull/185#discussion_r548356652) [](ancestors = 548356652)

Reply:
On a device, to check this behavior, I tried the below commands; here "DEVICE_METADATA" exists, without DEVICE_METADATA in config_db, hwsku will not be known so dockers will not come up. So instead of checking directly for "DEVICE_METADATA", I checked for some modified fields that will not exist.

>>> device_metadata = db_conn.get_all(db_conn.CONFIG_DB, "DEVICE_METADATA|loca")     
>>> print(device_metadata)
None
>>> device_metadata = db_conn.get_all(db_conn.CONFIG_DB, "DEVICE_METADAT")      
>>> print(device_metadata)
None
...
>>> hostname=device_metadata.get(b'hos')     
>>> print(hostname)                     
None

Reply:
The above comment does not hold good for SNMP docker with master branch with latest changes.
If a specific key does not exist in the table, A SWIG object is returned and None is not returned.
If key does not exist:
```
>>> from sonic_ax_impl import mibs
db_conn.connect(db_conn.CONFIG_DB)
>>> device_metadata = db_conn.get_all(db_conn.CONFIG_DB, "DEVICE_METADATA|loca")    
>>> print(device_metadata)
<swsscommon.swsscommon.FieldValueMap; proxy of <Swig Object of type 'swss::TableMap *' at 0x7f291e85a450> >
```
If DEVICE_METADATA exists, but key within DEVICE_METADATA table does not exist:
```
>>> device_metadata = db_conn.get_all(db_conn.CONFIG_DB, "DEVICE_METADATA|localhost")
>>> print(device_metadata)
<swsscommon.swsscommon.FieldValueMap; proxy of <Swig Object of type 'swss::TableMap *' at 0x7f291e85ab70> >
>>> print(device_metadata.get('foo'))
None
```